bokehjs/src/coffee/source/ajax_data_source.coffee

```diff
@@ -27,7 +27,8 @@ class AjaxDataSource extends RemoteDataSource.RemoteDataSource
       xhrField :
         withCredentials : true
       method : @get('method')
-      contentType : 'application/json'
+      contentType : @get('content_type')
+      headers : @get('http_headers')
```

Comment:
You could copy the general scheme of how I test the websocket client (fork a Python server). would have to write a cheesy little http server that echoes the headers and body it gets, or something along those lines.
Could test all aspects of Ajax source that way.